### PR TITLE
ui: add cluster info API

### PR DESCRIPTION
**Pull request by cuongdo**
Right now, only cluster ID is returned. In the future, we will likely
add other metadata, such as CockroachDB versions running in the cluster.

Resolves part of #5556 

cc @maxlang, the requester for this API

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5647)

### CODE REVIEW

**Comment by petermattis** [comment]
:lgtm:

---

Review status: 0 of 6 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5647#-:-KDyrbvYEBbUm3aZL2PF:1417088721)*

File: server/admin.go [hunk 1]
```diff
@@ -694,6 +697,17 @@ func (s *adminServer) GetUIData(_ context.Context, req *GetUIDataRequest) (*GetU
 	return resp, nil
 }
 
+// Cluster returns cluster metadata.
+func (s *adminServer) Cluster(_ context.Context, req *ClusterRequest) (*ClusterResponse, error) {
+	var resp ClusterResponse
```
Comment (tamird) [inline server/admin.go]: no point in constructing this until you're ready to return it

File: server/admin.proto [hunk 2]
```diff
@@ -225,6 +225,15 @@ message GetUIDataResponse {
   map<string, Value> key_values = 1;
 }
 
+// ClusterRequest requests metadata for the cluster.
+message ClusterRequest {
+}
+
+// ClusterResponse contains metadata for the cluster.
+message ClusterResponse {
+  string cluster_id = 1 [(gogoproto.customname) = "ClusterID"];
```
Comment (tamird) [inline server/admin.proto]: hm, could you customtype uuid.UUID as is done elsewhere? you'd save a string allocation, at least.

**Comment by cuongdo** [comment]
*[server/admin.proto, line 234 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5647#-KDyqmvNW1pfY2rzF_6w-r1-234:-KDz20R1yWSrQ18GT9IU:-604929958) ([raw file](https://github.com/cockroachdb/cockroach/blob/3924c3e7e31be89dc97bb5a14121ee3f27dfed7b/server/admin.proto#L234)):*
Is that really necessary? This API will be called extremely infrequently (once a day?). Plus, the `customtype` declaration seems to make the field a UUID, rather than the canonical string representation of the UUID. From an API point of view it's much nicer to have a hex string with some hyphens than a byte array.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5647)*

**Comment by tamird** [comment]
Reviewed 6 of 6 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[server/admin.proto, line 234 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5647#-KDyqmvNW1pfY2rzF_6w-r1-234:-KDz975_iOhiVDdNWyNU:352851982) ([raw file](https://github.com/cockroachdb/cockroach/blob/3924c3e7e31be89dc97bb5a14121ee3f27dfed7b/server/admin.proto#L234)):*
Yeah, fair enough.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5647)*

**Comment by cuongdo** [comment]
Review status: 5 of 6 files reviewed at latest revision, 1 unresolved discussion.

---

*[server/admin.go, line 702 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5647#-KDyqmvNW1pfY2rzF_6t-r1-702:-KE-6ym2-10-BjWztKGH:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/3924c3e7e31be89dc97bb5a14121ee3f27dfed7b/server/admin.go#L702)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5647)*

**Comment by tamird** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[server/admin.go, line 708 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5647#-KE-7KLWw2wfrJqJQ-A0:-KE-7KLWw2wfrJqJQ-A1:-1788251275) ([raw file](https://github.com/cockroachdb/cockroach/blob/4d3ca9d1dd136db2cdea276a282937534d2b1863/server/admin.go#L708)):*
why not return a literal?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5647#-:-KE-7NcmykfZ-MLw4W4i:1417088721)*

**Pull request merged**